### PR TITLE
Add anagram exercise

### PR DESCRIPTION
**Pull request by mfoda**
I'm a bit confused about what other files/folder I need to submit besides the ones for the exercise, test, and example implementation. Am I required to modify `config.json` and create a `.meta/tests.toml` file, and how would I go about doing that? Any pointers would be appreciated.

### CODE REVIEW

**Comment by mfoda** [comment]
Ok I think I figured it out and got everything in.

File: exercises/anagram/src/Example.hx [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+class Anagram {
+    public static function findAnagrams(subject: String, candidates: Array<String>): Array<String> {
+        var anagrams = [for (val in candidates) 
+                        if (subject.toLowerCase() != val.toLowerCase())
+                        if (sort_string(subject.toLowerCase()) == sort_string(val.toLowerCase()))
+                        val];
+        return anagrams;
+    }
+    
+    private static function sort_string(string:String): String {
```
Comment (davidbruce) [inline exercises/anagram/src/Example.hx:12]: Update sort_string to sortString and add a space between string:String for consistency.

File: exercises/anagram/src/Example.hx [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+        var _string = string.split("");
```
Comment (davidbruce) [inline exercises/anagram/src/Example.hx:13]: Rename `_string` to `chars` , `letters`, or something else to make it more easily identifiable as a it is a collection of characters.

**Review by davidbruce — CHANGES_REQUESTED** [review]
@mfoda 

Thank you for the submission!  I made a few comments on some changes I would like on the `Example.hx`.   They are just small nit picky things and overall it's a very clean solution.

**Comment by mfoda** [comment]
@davidbruce 
Thanks for the feedback! That's very helpful.

**Review by davidbruce — APPROVED** [review]
Looks good to me. :)

```
..............
Anagram
  no matches (Passed)
  detects two anagrams (Passed)
  does not detect anagram subsets (Passed)
  detects anagram (Passed)
  detects three anagrams (Passed)
  detects multiple anagrams with different case (Passed)
  does not detect non-anagrams with identical checksum (Passed)
  detects anagrams case-insensitively (Passed)
  detects anagrams using case-insensitive possible matches (Passed)
  detects anagrams using case-insensitive possible matches (Passed)
  does not detect an anagram if the original word is repeated (Passed)
  anagrams must use all letters exactly once (Passed)
  words are not anagrams of themselves (case-insensitive) (Passed)
  words other than themselves can be anagrams (Passed)
14 specs, 0 failures, 0 pending
```

**Pull request merged**